packages/nodes-base/nodes/N8nApi/CredentialDescription.ts

```diff
@@ -0,0 +1,160 @@
+import { INodeProperties } from 'n8n-workflow';
+import { parseAndSetBodyJson } from './GenericFunctions';
+
+export const credentialOperations: INodeProperties[] = [
+	{
+		displayName: 'Operation',
+		name: 'operation',
+		type: 'options',
+		noDataExpression: true,
+		default: 'create',
+		displayOptions: {
+			show: {
+				resource: ['credential'],
+			},
+		},
+		options: [
+			{
+				name: 'Create',
+				value: 'create',
+				action: 'Create a credential',
+				routing: {
+					request: {
+						method: 'POST',
+						url: '/credentials',
+					},
+				},
+			},
+			{
+				name: 'Delete',
+				value: 'delete',
+				action: 'Delete a credential',
+				routing: {
+					request: {
+						method: 'DELETE',
+						url: '=/credentials/{{ $parameter.credentialId }}',
+					},
+				},
+			},
+			{
+				name: 'Get Schema',
+				value: 'getSchema',
+				action: 'Get credential data schema for type',
+				routing: {
+					request: {
+						method: 'GET',
+						url: '=/credentials/schema/{{ $parameter.credentialTypeName }}',
+					},
+				},
+			},
+		],
+	},
+];
+
+const createOperation: INodeProperties[] = [
+	{
+		displayName: 'Name',
+		name: 'name',
+		type: 'string',
+		default: '',
+		required: true,
+		displayOptions: {
+			show: {
+				resource: ['credential'],
+				operation: ['create'],
+			},
+		},
+		routing: {
+			request: {
+				body: {
+					name: '={{ $value }}',
+				},
+			},
+		},
+		description: 'Name of the new credential',
+	},
+	{
+		displayName: 'Credential Type Name',
+		name: 'credentialTypeName',
+		type: 'string',
+		default: '',
+		required: true,
+		displayOptions: {
+			show: {
+				resource: ['credential'],
+				operation: ['create'],
+			},
+		},
+		routing: {
+			request: {
+				body: {
+					type: '={{ $value }}',
+				},
+			},
+		},
+		description: 'The name for the type of this credential',
+	},
+	{
+		displayName: 'Data',
+		name: 'data',
+		type: 'json',
+		default: '{}',
+		required: true,
+		typeOptions: {
+			alwaysOpenEditWindow: true,
+		},
+		displayOptions: {
+			show: {
+				resource: ['credential'],
+				operation: ['create'],
+			},
+		},
+		routing: {
+			send: {
+				// Validate that the 'data' property is parseable as JSON and
+				// set it into the request as body.data.
+				preSend: [parseAndSetBodyJson('data', 'data')],
+			},
+		},
+		description: 'JSON data for this credential',
+	},
+];
+
+const deleteOperation: INodeProperties[] = [
+	{
+		displayName: 'Credential ID',
+		name: 'credentialId',
+		type: 'string',
+		required: true,
+		default: '',
+		displayOptions: {
+			show: {
+				resource: ['credential'],
+				operation: ['delete'],
```

Comment:
Any pointers on how to reproduce this? I tried both deleting a credential I had created before as well as one that didn't exist (giving a 404), but never got the app into a wonky state.

<img width="917" alt="Screenshot 2022-09-13 at 17 51 58" src="https://user-images.githubusercontent.com/299644/189934390-ee7d08ce-1b4a-463a-bd8e-936e9079a9c6.png">
<img width="912" alt="Screenshot 2022-09-13 at 17 51 15" src="https://user-images.githubusercontent.com/299644/189934400-72fcf890-e6c9-4f0d-8ff4-824db5626abd.png">

Reply:
I did run n8n with tunnel on one machine and PR on other, after deleting instance on tunnel was unavailable with 404 error  and only restarting n8n fixed it, credentials was deleted though.

Reply:
and 404 it's not an error inside node, but whole site unavailable

Reply:
Okay, that's weird. Did the tunneled CLI show any error or stacktrace whatsoever (or can we get the logs somewhere), or just stopped responding until restart?

I've tried running this locally with and without tunnel (localhost:5678 and tunnel address) from within the same monorepo (cli and editor-ui in the same branch), and tried creating and deleting n8nApi credentials, and also removing the credential I was currently using for the call.

Unfortunately, couldn't get anything to break. Both the app and API would stay functional for me. I'm thinking what breaks for you might be unrelated to how this node works specifically. It would be good to understand the root cause, though. Could you maybe create a new bug issue for this, and we could try and tackle it separately?

Reply:
Sorry @mieky , it was something wrong with setup on my side, tried today with two different browsers and can not reproduce issue